src/lexer/lexer.sml

```diff
@@ -15,6 +15,7 @@ struct
         in
             ErrorMsg.reset();
             Newline.reset();
+            SrcComment.reset();
```

Comment:
I didn't notice a need to.

This was the cause of a very weird bug which took me a while to track down last night